Sources/System/Sockets/SocketDescriptor.swift

```diff
@@ -127,34 +153,113 @@ extension SocketDescriptor {
     @_alwaysEmitIntoClient
     public static var datagram: ConnectionType { ConnectionType(rawValue: _SOCK_DGRAM) }
 
-    /// Only available to the super user
+    /// Raw protocol interface. Only available to the super user
     ///
     /// The corresponding C constant is `SOCK_RAW`
     @_alwaysEmitIntoClient
     public static var raw: ConnectionType { ConnectionType(rawValue: _SOCK_RAW) }
+
+    /// Reliably delivered message.
+    ///
+    /// The corresponding C constant is `SOCK_RDM`
+    @_alwaysEmitIntoClient
+    public static var reliablyDeliveredMessage: ConnectionType {
+      ConnectionType(rawValue: _SOCK_RDM)
+    }
+
+    /// Sequenced packet stream.
+    ///
+    /// The corresponding C constant is `SOCK_SEQPACKET`
+    @_alwaysEmitIntoClient
+    public static var sequencedPacketStream: ConnectionType {
+      ConnectionType(rawValue: _SOCK_SEQPACKET)
+    }
+
+    public var description: String {
+      switch self {
+      case .stream: return "stream"
+      case .datagram: return "datagram"
+      case .raw: return "raw"
+      case .reliablyDeliveredMessage: return "rdm"
+      case .sequencedPacketStream: return "seqpacket"
+      default: return rawValue.description
+      }
+    }
   }
 
   /// TODO
   @frozen
-  public struct ProtocolID: RawRepresentable {
+  public struct ProtocolID: RawRepresentable, Hashable, CustomStringConvertible {
     @_alwaysEmitIntoClient
-    public let rawValue: CInt
+    public var rawValue: CInt
 
     @_alwaysEmitIntoClient
     public init(rawValue: CInt) { self.rawValue = rawValue }
 
+    @_alwaysEmitIntoClient
+    public init(_ rawValue: CInt) { self.rawValue = rawValue }
+
     /// The default protocol for the domain and connection type combination.
     @_alwaysEmitIntoClient
-    public static var `default`: ProtocolID { self.init(rawValue: 0) }
+    public static var `default`: ProtocolID { Self(0) }
+
+    /// Internet Protocol (IP)
```

Comment:
I think providing anything here is a *serious* mistake, other than arguably `ICMP`. If someone is doing something super obscure, they can use the raw value initializer.

Just because the headers conflated the `#define`s for completely different semantic purposes, doesn't mean System should too.

*If* we actually are going to do this, then we should unify on this `struct` and use it inside the addresses as well, in which case we thrust historical baggage on callers to `socket.open`, and explicitly warn against anything other than `default` in the docs.

Reply:
Resolution: we will put the constants here, but won't print them out. `Option.Level` will remain distinct. `Option` likewise won't be printed out.